### PR TITLE
V3—graph-responsive-layout

### PR DESCRIPTION
**Pull request by bfinzer**
* Got rid of hardcoded margin
* GraphLayout receives desired extents from axes and legend. These are used to compute bounds for each piece of the component
* Each piece of the graph requests its computed bounds from the GraphLayout and transforms accordingly

### CODE REVIEW

File: v3/src/components/graph/components/background.tsx [hunk 1]
```diff
@@ -58,18 +56,19 @@ export const Background = forwardRef<SVGGElement, IProps>((props, ref) => {
     previousMarqueeRect = useRef<rTreeRect>(),
 
     onDragStart = useCallback((event:{ x: number; y: number; sourceEvent: {shiftKey: boolean} }) => {
+      const plotBounds = layout.computedBounds.get('plot') as Bounds
```
Comment (kswenson) [inline v3/src/components/graph/components/background.tsx:59]:
```suggestion
      const { computedBounds } = layout,
      plotBounds = computedBounds.get('plot') as Bounds
```
and update the dependencies to just `layout`.

**Review by kswenson — APPROVED** [review]
👍 Looks good! Just the usual assortment of minor suggestions

File: v3/src/components/graph/hooks/use-axis.ts [hunk 2]
```diff
@@ -86,37 +152,72 @@ export const useAxis = ({axisModel, axisElt, showGridLines, enableAnimation}: IU
...
+        (enter) => {
+          enter.append('text')
+            .attr('class', 'axis-title')
+            .attr('text-anchor', 'middle')
+            .attr('data-testid', `axis-title-${place}`)
+        })
+
+  }, [axisElt, halfRange, label, place, titleRef])
 
+  // update on component refresh Todo: necessary?
```
[the remark this answers is not in the document]
Comment (bfinzer) [reply v3/src/components/graph/hooks/use-axis.ts:218]: Tried commenting it out and immediately the graph wasn't keeping itself up to date.

File: v3/src/components/graph/hooks/use-axis.ts [hunk 3]
```diff
@@ -1,69 +1,135 @@
...
 
+  const getLabelBounds = (s = 'Wy') => {
+      const textElement = selection().append('text').attr('y', 500),
+        bounds = textElement.text(s).node()?.getBoundingClientRect() || {left: 0, top: 0, width: 100, height: 20}
+      textElement.remove()
+      return bounds
+    }
+
+  const computeDesiredExtent = useCallback(() => {
+    const labelHeight = getLabelBounds(label || 'Wy').height
```
[the remark this answers is not in the document]
Comment (bfinzer) [reply v3/src/components/graph/hooks/use-axis.ts:53]: yup

**Pull request merged**